src/librustc_lint/unused.rs

```diff
@@ -355,10 +355,10 @@ impl EarlyLintPass for UnusedParens {
         let (value, msg, followed_by_block) = match e.node {
             If(ref cond, ..) => (cond, "`if` condition", true),
             While(ref cond, ..) => (cond, "`while` condition", true),
-            IfLet(_, ref cond, ..) => (cond, "`if let` head expression", true),
-            WhileLet(_, ref cond, ..) => (cond, "`while let` head expression", true),
-            ForLoop(_, ref cond, ..) => (cond, "`for` head expression", true),
-            Match(ref head, _) => (head, "`match` head expression", true),
+            IfLet(_, ref cond, ..) => (cond, "`if let` scrutinee", true),
+            WhileLet(_, ref cond, ..) => (cond, "`while let` scrutinee", true),
+            ForLoop(_, ref cond, ..) => (cond, "`for` scrutinee", true),
```

Comment:
I'm not sure "scrutinee" is correct here. The obvious "iterator expression" isn't really correct either, since it only has to implement `IntoIterator`. I guess this is fine for now.